src/Microsoft.Azure.SignalR.Common/Auth/JwtPayload.cs

```diff
@@ -0,0 +1,129 @@
+                this[JwtRegisteredClaimNames.Iat] = EpochTime.GetIntDate(issuedAt.Value.ToUniversalTime());
+
+            if (!string.IsNullOrEmpty(issuer))
+                this[JwtRegisteredClaimNames.Iss] = issuer;
+
+            // if could be the case that some of the claims above had an 'aud' claim;
+            if (!string.IsNullOrEmpty(audience))
+                AddClaim(new Claim(JwtRegisteredClaimNames.Aud, audience, ClaimValueTypes.String));
+        }
+
+        // Copied from https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/6.22.0/src/System.IdentityModel.Tokens.Jwt/JwtPayload.cs#L474
+        public void AddClaim(Claim claim)
+        {
+            if (claim == null)
+            {
+                throw LogHelper.LogExceptionMessage(new ArgumentNullException("claim"));
+            }
+
+            AddClaims(new Claim[1] { claim });
+        }
+
+        // Modified from https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/6.22.0/src/System.IdentityModel.Tokens.Jwt/JwtPayload.cs#L489 <summary>
+        // Modification:
+        //     object jsonClaimValue = claim.ValueType.Equals(ClaimValueTypes.String, StringComparison.Ordinal) ? claim.Value : TokenUtilities.GetClaimValueUsingValueType(claim);
+        public void AddClaims(IEnumerable<Claim> claims)
+        {
+            if (claims == null)
+            {
+                throw LogHelper.LogExceptionMessage(new ArgumentNullException(nameof(claims)));
+            }
+
+            foreach (Claim claim in claims)
+            {
+                if (claim == null)
+                {
+                    continue;
+                }
+
+                string jsonClaimType = claim.Type;
+                object jsonClaimValue = claim.ValueType.Equals(ClaimValueTypes.String, StringComparison.Ordinal) ? claim.Value : TokenUtilities.GetClaimValueUsingValueType(claim);
+                object existingValue;
+
+                // If there is an existing value, append to it.
+                // What to do if the 'ClaimValueType' is not the same.
+                if (TryGetValue(jsonClaimType, out existingValue))
+                {
+                    IList<object> claimValues = existingValue as IList<object>;
+                    if (claimValues == null)
+                    {
+                        claimValues = new List<object>();
+                        claimValues.Add(existingValue);
+                        this[jsonClaimType] = claimValues;
+                    }
+
+                    claimValues.Add(jsonClaimValue);
+                }
+                else
+                {
+                    this[jsonClaimType] = jsonClaimValue;
+                }
+            }
+        }
+
+        // Simplified from https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/6.22.0/src/System.IdentityModel.Tokens.Jwt/JwtPayload.cs#L761
+        public string Base64UrlEncode()
```

Comment:
looks like duplicate with the one in JwtHeader? maybe have a EncodeJson in Base64UrlEncoder?

Reply:
In original JWT package [Microsoft.IdentityModel.JsonWebTokens](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/tree/6.22.0), there are duplicate method `Base64UrlEncode` in `JwtHeader` and `JwtPayload`. Simply follow the original code to make our code maintainable in the future